src/lang/Calc.g4

```diff
@@ -46,12 +47,16 @@ expression
    | 'let' declarationList 'in' expressionList 'end'                                # LetExpression
    | '(' expression ')'                                                             # ParenthesizedExpression
    | '(' expressionList ')'                                                         # SequenceExpression
+   | '[' listElements ']'                                                           # ListExpression
+   | NIL                                                                            # EmptyListExpression
    ;
 
 lambda : 'fn' pattern '=>' expression;
 
 expressionList : expression (';' expression)*;
 
+listElements : expression (',' expression)*;
```

Comment:
Can we move this into ListExpression rule above directly?

Reply:
Done.